### DISCUSSION TITLE
how to convert YUV4:4:4 to AYUV?

### DISCUSSION THREAD

**Discussion by qihaoliu0** [ImageMagick/ImageMagick#5446] [Help] [answered]
I have an image encoding as YUV4:4:4 and using 8-bits/channel, how can i convert it to a new format by using imagemagick, which adds an alpha channel and still keeps the YUV data？

**Comment by snibgo** [comment]
What version of IM, on what platform? What image format is the input (TIFF, PNG, etc)? What is the desired output format?

It would help if you supplied a sample input file.

**Reply by qihaoliu0** [reply]
Thanks for u reply, this is my IM && OS version.

```
IM: ImageMagick 6.9.11-60 Q16 x86_64
OS: Debian GNU/Linux 11 (bullseye)
Linux-Kernel: Linux 5.10.16.3-microsoft-standard-WSL2 
```

I use OpenCV to convert an RGB image to a YUV4:4:4 Planar format(YYYY..UUUU...VVV) by the code below.

```python
import  numpy as np
import cv2 as cv

path =r"rgb.png"
bgr_data = cv.imread(path, -1)
bgr_data = np.uint8(bgr_data) #Y/U/V is 8-bits
img_yuv = cv.cvtColor(bgr_data, cv.COLOR_BGR2YUV)
img_yuv_planner = np.moveaxis(img_yuv, -1, 0)
img_yuv_planner.tofile(r"yuv_444.yuv")
```

My input format is YUV4:4:4 with Planar encode(YYY..UUU..VVV..), each Y/U/V costs 8bits, I want to use ImageMagick to convert this kind of format to a YUV+Alpha format(or other formats can represent YUV+alpha format). each Y/U/V/A still cost 8-bits.

> I find a format called YCbCrA in [here](https://imagemagick.org/script/formats.php). but a don't know how to encode the YUV+Alpha image like YUV Planar(YY..UUU...VVV...) format.

Thanks!


> P.s.
>
> This is my RGB png, it can use python code forward to convert. 
>
> ![rgb](https://user-images.githubusercontent.com/100996669/184810832-b21238fc-6c95-47a8-83e2-f6694c19998f.png)

**Comment by snibgo** [comment, marked as answer]
So your file format is headerless, 8 bits per channel per pixel, with all the Y values first, then all the U values, then all the V values. If you have `N` pixels, your input file has `3*N` bytes. You want an output file that has an extra `N` bytes added to the end for the alpha channel, so it has `4*N` bytes. Each extra byte should be 255.

Is that correct?

If so, then the obvious solution is to put that in your code that generates the file. Another solution would be a trivial C program that adds the bytes. I expect it can be done with bash utilities, perhaps with `awk` or something.

It can be done with IM like this:
```
magick -interlace Plane -depth 8 -size 267x233 RGB:x.bin -alpha set RGBA:x1.bin
```
For IM v6, use `convert` instead of `magick`. Instead of `267x233`, use the correct width and height for your image.

We are reading the bytes as if they represented RGB instead of YUV. As we are doing no processing with these values, it doesn't matter what they represent.

**Reply by qihaoliu0** [reply]
Thanks! That's what right now I needed.
